src/Packages/Marketplace/OnRamp/OnRamp.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Cysharp.Threading.Tasks;
+using UnityEngine;
+
+namespace Immutable.Marketplace.OnRamp
+{
+    public class OnRamp
+    {
+        private readonly Passport.Passport _passport;
+        private static readonly Dictionary<string, string> TransakBaseUrls = new Dictionary<string, string>
+        {
+            { "sandbox", "https://global-stg.transak.com" },
+            { "production", "https://global.transak.com/" }
+        };
+
+        private static readonly Dictionary<string, string> TransakApiKeys = new Dictionary<string, string>
+        {
+            { "sandbox", "d14b44fb-0f84-4db5-affb-e044040d724b" }, // This can be hardcoded as it is a public API key
+            { "production", "ad1bca70-d917-4628-bb0f-5609537498bc" }
+        };
+
+        public OnRamp(Passport.Passport passport)
+        {
+            _passport = passport;
+        }
+
+        public async UniTask<string> GetLink(
+    string fiatCurrency = "USD",
+    string defaultFiatAmount = "50",
+    string defaultCryptoCurrency = "IMX",
+    string networks = "immutablezkevm",
+    bool disableWalletAddressForm = true)
+        {
+            await _passport.ConnectImx();
```

Comment:
To on the ramp, they need to be connected to Immutable X?

What happens if the user is already connected to Passport before calling this function? 
* passport.Connectmx()
* onRamp.GetLink(...)

Will this open the browser and take them through the login flow again?

Reply:
Is this line the main reason the Passport instance needs to be passed into this function?